concrete_biopython/FheSeq.py

```diff
@@ -68,13 +252,43 @@ def __init__(self, data, length=None):
                 self._data = data._data[:].reshape(1)  # take a copy
             else:
                 self._data = fhe.zeros(0) 
+        # TODO: add possibility for clear data, when concrete allows operations on clear Tracers
+        # elif isinstance(data, str):
+        #     # convert str to clear integers
+        #     self._data = self._seq_interface.to_integers(data)
+        # elif isinstance(data, np.ndarray):
+        #     # check array type
+        #     if data.dtype != np.int:
+        #         raise ValueError("data is an ndarray but doesn't have integer as it's dtype")
+        #     # store clear integers
+        #     self._data = data.copy()
         else:
             raise TypeError(
-                "data should be either a concrete.fhe.tracing.tracer.Tracer object or a _FheSeqAbstractBaseClass object"
-            )    
+                "data should be either of these types: \n\
+                \t- concrete.fhe.tracing.tracer.Tracer\n\
+                \t- _FheSeqAbstractBaseClass\n\
+                "
+            )
+                # \t- str\n\
+                # \t- numpy.ndarray (with dtype = np.int)\n\
+                # "
+            #)   
+
+        if seq_interface is None:
+            raise ValueError("seq_interface is None")
 
+        self._seq_interface = seq_interface
+
+    def get_seq_interface(self):
+        """
+        seq_interface getter
+        """
+        return self._seq_interface
```

Comment:
We're already in a `Seq` class, it's better to call this `seq.interface()`, and maybe make it a proper getter using `@property`